### DISCUSSION TITLE
Capturing Packets in a Kubernetes Environment with Containerd

### DISCUSSION THREAD

**Discussion by lcc3108** [gojue/ecapture#597] [Q&A] [answered]
Hello

I would like to check the traffic of the pod running through containerd through that tool. 

https://github.com/gojue/ecapture/issues/65

The above article was written based on docker, so I used it like the command below in accordance with containerd.

```
ecapture tls --libssl=/run/containerd/io.containerd.runtime.v2.task/k8s.io/${container_id}/rootfs/lib64/libssl.so.1.1
```

When executing curl in pod with this command, it is captured well, but the server's communication is not captured.

I saw what socket the server uses as lsof etc, but I'm using java, so I can't find any specific files.

Do you happen to have any instructions to find out what libssl is used in java, node, etc? 

Thank you.

**Comment by cfc4n** [comment, marked as answer]
The program written in `Java` does not utilize the libssl library. `Node` is **statically compiled** and integrated into the executable file. 

Therefore, it is impossible to intercept encrypted communications of the `Java` program. However, support for `Node` encryption can be provided; you need to set the `Node` file with the `--libssl` argument.

**Reply by lcc3108** [reply]
I'm gonna have to find another way for java

In the case of the node, as instructed, the node itself is designated as libssl, so it works well.

Thank you for your answer.
